oauthenticator/gitlab.py

```diff
@@ -168,25 +168,25 @@ class GitLabOAuthenticator(OAuthenticator):
             return None
 
 
-    async def _check_group_whitelist(self, user_id, access_token):
+    async def _check_group_whitelist(self, username, access_token):
         http_client = AsyncHTTPClient()
         headers = _api_headers(access_token)
         # Check if user is a member of any group in the whitelist
         for group in map(url_escape, self.gitlab_group_whitelist):
-            url = "%s/groups/%s/members/%d" % (GITLAB_API, group, user_id)
+            url = "%s/groups/%s/members/all?query=%s" % (GITLAB_API, group, username)
```

Comment:
Oh, and I believe we need to encode at least `group` here and `project` below, as these these can contain `/` (e.g. the project `gitlab-org/gitlab` 😉 ) which need to be encoded as `%2F`.

Reply:
I was just blind. Missed the `map(url_escape, ...)` part in the line above